### PR TITLE
refactor: improve shares requests

### PR DESCRIPTION
**Pull request by SeanJCasey**

### CODE REVIEW

File: src/fund/shares/Shares.sol [hunk 1]
```diff
@@ -59,45 +52,48 @@ contract Shares is IShares, TokenUser, Spoke, SharesToken {
...
+        uint256 _minSharesQuantity
     )
         external
         override
         onlySharesRequestor
-        returns (uint256 costInInvestmentAsset_)
+        returns (uint256 sharesBought_)
     {
-        costInInvestmentAsset_ = getSharesCostInAsset(_sharesQuantity, _investmentAsset);
+        address denominationAsset = DENOMINATION_ASSET;
```
Comment (travs) [inline src/fund/shares/Shares.sol]: this is fine, but do we really need to assign from the constant?

File: src/requests/SharesRequestor.sol [hunk 2]
```diff
@@ -117,158 +110,155 @@ contract SharesRequestor is DSMath, TokenUser, AmguConsumer {
...
-        // Do the asset transfers in a separate step after altering state
-        __transferRequestSurplusAssets(
-            _requestOwner,
-            request.investmentAsset,
-            request.maxInvestmentAmount,
-            investmentAmountFilled,
-            request.incentiveFee
-        );
+        // Reward sender with incentive
+        msg.sender.transfer(request.incentiveFee);
```
Comment (travs) [inline src/requests/SharesRequestor.sol:124]: Maybe I'm missing some permissioning but this is susceptible to frontrunning. I guess we will fix that later on?

**Review by travs — CHANGES_REQUESTED** [review]
Looks good, just those couple notes

File: src/requests/SharesRequestor.sol (hunk 2 above)
Comment (SeanJCasey) [reply src/requests/SharesRequestor.sol:124]: As discussed in our chat, we'll leave it as-is as this is by design, but we will want to revisit that design before the next release.

File: src/fund/shares/Shares.sol (hunk 1 above)
Comment (SeanJCasey) [reply src/fund/shares/Shares.sol]: Fixed

**Review by travs — APPROVED** [review]

**Pull request merged**